addons/binding/org.openhab.binding.icloud/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="icloud" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+	<bridge-type id="account">
+		<label>iCloud Account</label>
+		<description>Provides various information for iCloud devices registered with this account.</description>
+
+		<config-description>
+			<parameter name="appleId" type="text" required="true">
+				<context>email</context>
+				<label>Apple Id</label>
+				<description>Apple Id (e-mail) to access iCloud information.</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<context>password</context>
+				<label>Password</label>
+				<description>Apple iCloud password for the given Id.</description>
+			</parameter>
+			<parameter name="refreshTimeInMinutes" type="integer" min="5" max="65535" unit="min">
+				<label>Refresh</label>
+				<description>Refresh time in minutes.</description>
+				<default>5</default>
+			</parameter>
+		</config-description>
+	</bridge-type>
+
+	<thing-type id="device">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+
+		<label>iCloud Device</label>
+
+		<channels>
+			<channel typeId="BatteryStatus" id="batteryStatus" />
+			<channel typeId="system.battery-level" id="batteryLevel" />
+			<channel typeId="FindMyPhone" id="findMyPhone" />
+			<channel typeId="Location" id="location" />
+			<channel typeId="LocationAccuracy" id="locationAccuracy" />
+			<channel typeId="LocationLastUpdate" id="locationLastUpdate" />
+			<channel typeId="DeviceName" id="deviceName" />
+		</channels>
+
+		<representation-property>deviceId</representation-property>
+
+		<config-description>
+			<parameter name="deviceId" type="text" required="true">
+				<label>Device id</label>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+	
+	<channel-type id="Refresh">
+		<item-type>Switch</item-type>
+		<label>Refresh</label>
+		<description>Update all devices registered with this account immediately.</description>
+	</channel-type>
+	<channel-type id="BatteryStatus">
+		<item-type>String</item-type>
+		<label>Battery Status</label>
+	    <state>

```

Comment:
please add `readOnly="true"` to the state as otherwise the UIs allow changing the value through a selection box.

Reply:
Also please pay attention to tabs vs. spaces

![grafik](https://user-images.githubusercontent.com/2870104/34003643-1c80bab0-e0f6-11e7-9c63-d80226a548dd.png)

Btw could you please tell me which editor you are using? Mixing tabs and spaces in one line is just cruel 😨

Reply:
Simply apply the automatic code formatter through Ctrl+Shift+F!

Reply:
@kaikreuzer You should introduce "Format on Save (changed lines)" in the project settings. This is a ever-reoccurring annoyance :)

Reply:
Its configured for Java code but I suppose not (yet) for XML?

Reply:
Done,

Reply:
Afair, there wasn't a possibility to activate that for XMLs, while we have it in place for Java files...